pollux/core/src/main/scala/org/hyperledger/identus/pollux/core/service/uriResolvers/DidUrlResolver.scala

```diff
@@ -1,17 +1,112 @@
+import org.hyperledger.identus.pollux.vc.jwt.*
+import org.hyperledger.identus.shared.crypto.Sha256Hash
 import org.hyperledger.identus.shared.http.{GenericUriResolverError, UriResolver}
+import org.hyperledger.identus.shared.models.StatusCode
 import zio.*
 
 class DidUrlResolver(httpUrlResolver: HttpUrlResolver, didResolver: DidResolver) extends UriResolver {
+  import DidUrlResolver.*
+
   def resolve(uri: String): IO[GenericUriResolverError, String] = {
-    ??? // TODO: implement did URL resolver
+
+    for {
+      parsed <- ZIO.fromTry(Url.parseTry(uri)).mapError(_ => InvalidURI(uri))
+      maybeResourceService = parsed.query.param("resourceService")
+      maybeResourcePath = parsed.query.param("resourcePath")
+      maybeResourceHash = parsed.query.param("resourceHash")
+      serviceAndPath <- ZIO
+        .fromOption(maybeResourceService zip maybeResourcePath)
+        .mapError(_ => MissingRequiredParams(uri))
+      (resourceService, resourcePath) = serviceAndPath
+      didStr = parsed.removeQueryString().toString
+      didDocument <- didResolver.resolve(didStr).flatMap {
+        case DIDResolutionFailed(err) =>
+          err match
+            case InvalidDid(message)                 => ZIO.fail(DidResolutionError(didStr, message))
+            case NotFound(message)                   => ZIO.fail(DidResolutionError(didStr, message))
+            case RepresentationNotSupported(message) => ZIO.fail(DidResolutionError(didStr, message))
+            case InvalidPublicKeyLength(message)     => ZIO.fail(DidResolutionError(didStr, message))
+            case InvalidPublicKeyType(message)       => ZIO.fail(DidResolutionError(didStr, message))
+            case UnsupportedPublicKeyType(message)   => ZIO.fail(DidResolutionError(didStr, message))
+            case jwt.Error(error, message)           => ZIO.fail(DidResolutionError(didStr, message))
+        case DIDResolutionSucceeded(didDocument, didDocumentMetadata) => ZIO.succeed(didDocument)
+      }
+      service <- ZIO
+        .fromOption(
+          didDocument.service.find(x => x.id == s"$didStr#$resourceService" && x.`type` == "LinkedResourceV1")
+        )
+        .mapError(_ =>
+          DidDocumentParsingError(
+            s"""Service with id: "$resourceService" and type: "LinkedResourceV1" not found inside DID document"""
+          )
+        )
+      baseUrl <- ZIO
+        .fromOption(service.serviceEndpoint.asString)
+        .mapError(_ => DidDocumentParsingError("serviceEndpoint is not a string"))
+
+      path <- ZIO.fromOption(UrlPath.parseOption(resourcePath)).mapError(_ => InvalidUrlPath(resourcePath))
+      finalUrl <- ZIO
+        .fromTry(Url.parseTry(baseUrl).map(x => x.withPath(path)).map(_.toString))
+        .mapError(_ => InvalidURI(baseUrl))
+      result <- httpUrlResolver.resolve(finalUrl)
+
+      validatedResult <- maybeResourceHash.fold(ZIO.succeed(result)) { hash =>
+        val computedHash = Sha256Hash.compute(result.getBytes()).hexEncoded
```

Comment:
I think this may be hashing the wrong value
If I am not mistakne, the content of `result` in your code is a JSON with the form

```
 { 
    “resource“ : <base64url string> 
 }
```
while we want to hash the value in the `"resource"` field